src/apps/answers/service.py

```diff
@@ -578,6 +594,69 @@ async def get_activity_answer(
 
+    async def _validate_user_role_for_take_now(self, applet_id: uuid.UUID, respondent: SubjectSchema) -> None:
+        is_user_role_restrictive = await UserAppletAccessCRUD(self.session).get_by_roles(
+            respondent.user_id,
+            applet_id,
+            # Define a list of roles prohibited from accessing the applet
+            [Role.EDITOR, Role.COORDINATOR, Role.REVIEWER],
+        )
+
+        if is_user_role_restrictive:
+            raise MultiinformantAssessmentNoAccessApplet(
+                "Access denied: User lacks the required ownership or managerial role for this operation."
+            )
+
+        return None
+
+    async def _validate_relation_between_subjects_in_applet(
+        self, respondent_subject: SubjectSchema, target_subject: SubjectSchema, source_subject: SubjectSchema
+    ) -> None:
+        """
+        Validate the relationship between subjects in an applet.
+        - respondent_subject: the subject inputting the answers.
+        - target_subject: the subject about whom the responses are.
+        - source_subject: the subject providing the responses.
+
+        This method checks:
+        1. If there is a relationship between the logged-in user (respondent_subject) and target_subject.
+        2. If there is a relationship between the respondent_subject and the source_subject.
+        Raises:
+            MultiinformantAssessmentNoAccessApplet: If no valid relationship is found.
+        """
+
+        if respondent_subject.id != source_subject.id:
+            relation_respondent_source_subjects = await SubjectsCrud(self.session).get_relation(
+                respondent_subject.id, source_subject.id
+            )
+            if not relation_respondent_source_subjects or (
+                relation_respondent_source_subjects
+                and self._check_subject_relation_is_temporary_and_expired(relation_respondent_source_subjects)
+            ):
+                raise MultiinformantAssessmentNoAccessApplet("Subject relation not found")
+
+        if respondent_subject.id != target_subject.id:
+            relation_respondent_target_subjects = await SubjectsCrud(self.session).get_relation(
+                respondent_subject.id, target_subject.id
+            )
+            if not relation_respondent_target_subjects or (
+                relation_respondent_target_subjects
+                and self._check_subject_relation_is_temporary_and_expired(relation_respondent_target_subjects)
+            ):
+                raise MultiinformantAssessmentNoAccessApplet("Subject relation not found")
+
+        return None
+
+    def _check_subject_relation_is_temporary_and_expired(relation: SubjectRelation) -> bool:
+        if relation.relation != "take-now" or relation.meta is None:
+            return False
+
+        expires_at = datetime.fromisoformat(relation.meta["expiresAt"])
+        if expires_at > datetime.now():
+            return False
+
+        return True
```

Comment:
We can remove this one in favour of `is_take_now_relation` and `is_valid_take_now_relation`